src/Form/Form.tsx

```diff
@@ -347,18 +350,10 @@ const Form: FormComponent = React.forwardRef((props: FormProps, ref: React.Ref<F
     onSubmit?.(checkStatus, event);
   });
 
-  const handleFieldError = useEventCallback((name: string, errorMessage: React.ReactNode) => {
-    const formError = {
-      ...realFormError,
-      [name]: errorMessage
-    };
-    setFormError(formError);
-    onError?.(formError);
-    onCheck?.(formError);
-  });
-
-  const handleFieldSuccess = useEventCallback((name: string) => {
-    removeFieldError(name);
+  const handleErrorChange = useEventCallback((nextError: Record<string, ReactNode>) => {
+    setFormError(nextError);
+    onError?.(nextError);
```

Comment:
Check whether the Error object has changed, if not, do not call `onError`.

```tsx
if (!shallowEqual(realFormError, nextError)) {
  onError?.(nextError);
}
```

Reply:
done